eduvpn/i18n.py

```diff
@@ -9,9 +13,82 @@
 country_mapping = None
 
 
+def init(lets_connect: bool, prefix: str):
+    """
+    Init locale and gettext, returns text domain
+    """
+    domain = 'LetConnect' if lets_connect else 'eduVPN'
+    directory = os.path.join(prefix, 'share/locale')
+
+    locale.setlocale(locale.LC_ALL, '')
+    locale.bindtextdomain(domain, directory)  # type: ignore
+    locale.textdomain(domain)  # type: ignore
+    install(domain, localedir=directory)
+
+    return domain
+
+
+def f(fstring: str) -> str:
```

Comment:
Nice hack, but IMO it's too much magic for too little gain.
It's used only in one single function, so doesn't even pass the threshold of reducing total lines of code.
Better to be explicit: `_('Hey, {name}', name=username)` and similar for plurals.

Reply:
Since I am also a translator of the Windows version, the function is oriented to their specifications.
This similarity may make it somewhat easier for the translators.
Quote from a post by Simon:

> In order to translate "Valid for 3 minutes and 1 second" a pluralizable placeholders like {number:singular|plural} were introduced in English texts.
> 
> Here is how to translate "{0:second|seconds}"...
> 
> The number 0 in the placeholder represents zero-based index of the parameter to be inserted. Do not change it in translation. Indexes of parameters are described in developer notes of each string to translate.
> 
> Add semicolon : and required number of plural forms separated by vertical line |. Number of plural forms is language dependant. We are using gettext definition of number and order of plural forms described here: http://docs.translatehouse.org/projects/localization-guide/en/latest/l10n/pluralforms.html.
> 
> Most of the languages have only two (2) forms of plural: singular|plural.
> 
> Japanese does not distinct between singular and plural, therefore there is only one form of plural in Japanese: {0:秒}
> German translation would be something like {0:Sekunde|Sekunden} (singular|plural)
> Slovenian translation is {0:sekunda|sekundi|sekunde|sekund} (singular|dual|plural 3-4|plural ≥5)
> Ukrainian translation would be like {0:секунда|секунди|секунд} (singular|plural 2-4|plural ≥5)
> 
> If you have any question about this, please do not hesitate to ask me.